### ISSUE TITLE
Expose alternative `InputBlur`-like event for use with `useMultipleSelection`

### ISSUE THREAD

**Issue by bnjmnt4n** [downshift-js/downshift#1159]
- `downshift` version: v6

Feature request:

downshift [attaches event listeners](https://github.com/downshift-js/downshift/blob/ba0ec8a51f3ee60d52b90122fc3efa7e26532160/src/hooks/utils.js#L305-L382) to the document to trigger the `InputBlur` event when the the user clicks outside of the downshift menu element when the menu is open. I'd like to request a similar event to be triggered for use with `useMultipleSelection`.

Here's my usecase: I have a multiple selection with different styles for the currently active selected item. I'd like to be able to reset this when the user clicks outside the menu, so anytime the user clicks outside of the menu elements I'd call something like `setActiveIndex(-1)`. I feel that this is a common enough use case that this could be integrated within `useMultipleSelection`.

**Comment by silviuaavram** [comment]
Hi! `InputBlur` is triggered when you blur on the combobox/select, it's not related to the selected items from useMultipleSelection. Probably you have this in mind as well.

What you want is `SelectedItemBlur` that triggers when you blur from the selected item, but only on Tab / click/tap outside??

**Comment by bnjmnt4n** [comment]
> What you want is `SelectedItemBlur` that triggers when you blur from the selected item, but only on Tab / click/tap outside?

Yes, this would be exactly what I'm asking for. Perhaps some code that checks if `activeIndex !== -1` and resets it to `-1` when the user clicks outside of the active items.

**Comment by silviuaavram** [comment]
Actually I don't think this should be done. The reasons is that we should keep tabindex=0 on the active item for you to be able to return to it via Tab.

**Comment by bnjmnt4n** [comment]
I think this is a pretty common use case for multiselection components. For example, Gmail has this behaviour as well, where if you've selected an email and click elsewhere, it deactivates the selection. I agree that this might not be a good default for everyone, which is why adding an extra event like `SelectedItemBlur` can come in handy so users who do want to implement this behaviour can use a state reducer or event handler. Otherwise, it would be possible to implement this outside of downshift, but it would require duplicating a lot of the work that could already be done by downshift (which keeps track of the element refs and also has utilities in place for handling clicks outside of the elements).
